paseos/visualization/space_animation.py

```diff
@@ -0,0 +1,280 @@
+        )  # if objects do not exist in known actors, remove from plot next update.
+        objects_to_add = list(
+            current_actors.difference(objects_in_plot)
+        )  # if known_actor does not exist in objects, add the actors and update in plot
+
+        plot_objects_to_remove = [
+            x for x in self.objects if x.actor in objects_to_remove
+        ]
+        for obj in plot_objects_to_remove:
+            obj.plot.trajectory.remove()
+            obj.plot.point.remove()
+            obj.plot.text.remove()
+
+        self.objects = [x for x in self.objects if x.actor not in objects_to_remove]
+
+        for obj_to_add in objects_to_add:
+            self.objects.append(DotMap(actor=obj_to_add))
+
+        # update positions of objects
+        for known_actor in current_actors:
+            for obj in self.objects:
+                if obj.actor == known_actor:
+                    pos, _ = known_actor.get_position_velocity(pk.epoch(local_time_d))
+                    pos_norm = [x / self._norm_coeff for x in pos]
+                    if "positions" in obj:
+                        obj.positions = np.vstack((obj.positions, pos_norm))
+                    else:
+                        obj.positions = np.array(pos_norm)
+        self._plot_actors()
+
+        # Step through trajectories to find max and min values in each direction
+        coords_max = [0, 0, 0]
+        coords_min = [0, 0, 0]
+        for obj in self.objects:
+            coords_max = np.maximum(obj.positions.max(axis=0), coords_max)
+            coords_min = np.minimum(obj.positions.min(axis=0), coords_min)
+        self.ax.set_xlim(coords_min[0], coords_max[0])
+        self.ax.set_ylim(coords_min[1], coords_max[1])
+        self.ax.set_zlim(coords_min[2], coords_max[2])
+
+        self.ax.set_title(self._sec_to_ddhhmmss(local_time_d / pk.SEC2DAY))
+
+        return
+
+    def _animate(self, sim: PASEOS, dt: float) -> List[Artist]:
+        """Advances the time of sim, updates the plot, and returns the axis objects
+        Args:
+        sim (PASEOS): simulation object.
+        dt: size of time step
+        Returns:
+            List[Artist]: list of Artist objects
+        """
+        sim.advance_time(dt)
+        self._update(sim)
+        return self.ax.get_children()
+
+    def _animation_wrapper(self, step: int, sim: PASEOS, dt: float) -> List[Artist]:
+        """wrapper to allow for frame numbers from animation
+        Args:
+        steps: current frame number of the animation
+        sim (PASEOS): simulation object.
+        dt: size of time step
+        Returns:
+            List[Artist]: list of Artist objects
+        """
+        return self._animate(sim, dt)
+
+    def animate(
+        self,
+        sim: PASEOS,
+        dt: float,
+        steps: int = 1,
+        name: str = None,
+        save_file: bool = False,
+    ) -> None:
+        """Animates paseos for a given number of steps with dt in each step.
```

Comment:
as above, this is important as otherwise sphinx will not be able to parse our docs